### ISSUE TITLE
Getting wrong UserID in PeoplePicker of @pnp/spfx-controls-react.

### ISSUE THREAD

**Issue by waqarjavaid25** [pnp/sp-dev-fx-controls-react#218]
When i use people picker i get following user in selectedItems function.

Output in Version:     **"@pnp/spfx-controls-react": "1.11.0",**
```
id: "i:0#.f|membership|user1@sharepointwaqar.onmicrosoft.com"
imageInitials: "WJ"
imageUrl: "https://outlook.office365.com/owa/service.svc/s/GetPersonaPhoto?email=user1@sharepointwaqar.onmicrosoft.com&UA=0&size=HR96x96"
optionalText: ""
secondaryText: "user1@sharepointwaqar.onmicrosoft.com"
tertiaryText: ""
text: "Waqar Javaid"
```

instead of 

Output in version `**"@pnp/spfx-controls-react": "1.6.0"**`
```
id: 10
imageInitials: ""
imageUrl: "https://outlook.office365.com/owa/service.svc/s/GetPersonaPhoto?email=user1@sharepointwaqar.onmicrosoft.com&UA=0&size=HR96x96"
optionalText: ""
primaryText: "Waqar Javaid"
secondaryText: "user1@sharepointwaqar.onmicrosoft.com"
selected: true
tertiaryText: ""
```

**Comment by sympmarc** [comment]
If you use
`ensureUser: true`
you'll get the ID from the User Information List instead of the account. If the user hasn't been added yet, this will ensure they are.

**Comment by waqarjavaid25** [comment]
Thanks Issue Resolved.
I want little explanation what do you mean by "user hasn't been added"

**Comment by sympmarc** [comment]
There's a User Information List in each Site Collection. It's a hidden list which contains the people who have ever visited or "touched" the Site Collection. When you create a new Site Collection, only you will be in the User Information List and your ID=1. Each person who visits that Site Collection is added to the User Information List, and that's where the ID you were looking for comes from. When you use the ensureUser option, the user you select is added to the User Information List (if they aren't already there) so you can get a valid ID for them.

**Comment by waqarjavaid25** [comment]
Dear @sympmarc ,

In my case, i have visited site multiple time and i am site collection administrator,so i am pretty sure my user is in user information list.
I think this is bug in new code version  because as per my original post, I got correct user id in version 1.6.0 and incorrect data in 1.11.0 version.

**Comment by coreyroth** [comment]
I just got bit by this after upgrading my PeoplePicker between versions.  I was writing the id field into a list item which broke when I upgraded.  Unfortunately, I didn't realize it before I had already deployed.

**Comment by AJIXuMuK** [comment]
Closing this one as old.
Please, feel free to open a new issue if you still experience problems with `PeoplePicker` in the latest version of the library.

**Comment by SreelekhaSrinivasan** [comment]
Found it. Took a while. I used PnP Spfx PeoplePicker in UI. The PeoplePicker site column in backend SPOnline list accepts the Id to resolve the user. 

We have the attribute ensureUser={true} present in the component. If the ensureUser={false} (or not mentioned) it retrieves the loginName again. 

Now after making the attribute to true, the json output is now retrieving like so,

id: 10      -------------------> integer value
imageInitials: ""
imageUrl: "https://outlook.office365.com/owa/service.svc/s/GetPersonaPhoto?email=user1@sharepointwaqar.onmicrosoft.com&UA=0&size=HR96x96"
optionalText: ""
primaryText: "Waqar Javaid"
secondaryText: "user1@sharepointwaqar.onmicrosoft.com"
selected: true
tertiaryText: ""
